Modules/@babylonjs/react-native/BabylonModule.ts

```diff
@@ -1,33 +1,72 @@
 import { NativeModules } from 'react-native';
-import { NativeEngine } from '@babylonjs/core';
+import { Engine, NativeEngine } from '@babylonjs/core';
 
-// This global object is part of Babylon Native.
+const disposedPropertyName = "BabylonReactNative_IsDisposed";
+
+declare const global: {
+    nativeCallSyncHook: any;
+};
+const isRemoteDebuggingEnabled = !global.nativeCallSyncHook;
+
+// This global object is owned by Babylon Native.
 declare const _native: {
     whenGraphicsReady: () => Promise<void>;
-    engineInstance: NativeEngine;
-}
+};
+
+// This JSI-based global object is owned by Babylon React Native.
+// This will likely be converted to a TurboModule when they are fully supported.
+declare const BabylonNative: {
+    readonly initializationPromise: Promise<void>;
```

Comment:
>initializationPromise [](start = 13, length = 21)

does this need to be a property or could it be a function? I have not seen great examples in react-native-windows for handing back non-value/non-object types as properties

Reply:
It could be a function if needed, it just didn't seem necessary. I would be surprised if this wasn't supported in react-native-windows, but if we find out its not, I will change it at that time to a function.

Reply:
I talked some with react-native-windows owners and if this is supposed to be a native module, this isn't supported. They also said that setEngineInstance isn't supported either as a function for a native module in UWP. We may want to limit the native module interface to the bare minimum and populate everything else with JSI/NAPI.

Reply:
Let's try to sync up on this tomorrow.

Reply:
Talked offline. I was confused here thinking this was a module definition when it will be a jsi/napi object declaration. This should work fine